CDP4Reporting/DataSource/IReportingDataSource.cs

```diff
@@ -28,26 +28,41 @@ namespace CDP4Reporting.DataSource
     using CDP4Common.EngineeringModelData;
     using CDP4Common.SiteDirectoryData;
 
+    using CDP4Dal;
+
     /// <summary>
     /// The interface used for creating a reporting data source.
     /// </summary>
-    /// <typeparam name="T">
-    /// The <see cref="ReportingDataSourceRow"/> representing the data source rows.
-    /// </typeparam>
-    public interface IReportingDataSource<T> where T : ReportingDataSourceRow, new()
+    internal interface IReportingDataSource
     {
         /// <summary>
         /// Creates a new data source instance.
         /// </summary>
-        /// <param name="option">
-        /// The <see cref="Option"/> for which the data source is built.
-        /// </param>
-        /// <param name="domainOfExpertise">
-        /// The <see cref="DomainOfExpertise"/> for which the data source is built.
-        /// </param>
         /// <returns>
-        /// A new <see cref="ReportingDataSourceClass{T}"/> instance.
+        /// An object instance.
         /// </returns>
-        ReportingDataSourceClass<T> CreateDataSource(Option option, DomainOfExpertise domainOfExpertise);
+        object CreateDataSource();
+    }
+
+    public abstract class ReportingDataSource : IReportingDataSource
```

Comment:
Indeed needs some finetuning. Multiple Constructors and Initialize/CreateDataSource overloads could be needed. It might even need a different way of adding parameters to stay backwards compatible in the future. Or maybe we need to create multiple abstract "default" implementations of IReportingDataSource. It's definitely not the final version of this mechanisme.